src/components/deploy/deploy_args.rs

```diff
@@ -63,24 +64,41 @@ impl From<SignerContext> for near_cli_rs::commands::ActionContext {
                     return Ok(prepopulated_transaction);
                 }
 
-                let call_result = network_config
-                    .json_rpc_client()
-                    .blocking_call_view_function(
-                        near_social_account_id,
-                        "get",
-                        serde_json::to_string(&crate::socialdb_types::SocialDbQuery {
-                            keys: local_widgets
-                            .keys()
+                let runtime = tokio::runtime::Builder::new_multi_thread().enable_all().build()?;
+                let chunk_size = 10;
+                let mut calls = Vec::with_capacity(local_widgets.len() / chunk_size + 1);
+                let local_widget_names = local_widgets.keys().cloned().collect::<Vec<_>>();
+
+                for (index, _) in local_widget_names.iter().enumerate().step_by(chunk_size) {
+                    let args = serde_json::to_string(&crate::socialdb_types::SocialDbQuery {
+                        keys: local_widget_names.iter().skip(index).take(chunk_size)
                             .map(|name| format!("{deploy_to_account_id}/widget/{name}/**"))
                             .collect(),
-                        })
-                        .wrap_err("Internal error: could not serialize SocialDB input args")?
-                        .into_bytes(),
-                        near_primitives::types::Finality::Final.into(),
-                    )
-                    .wrap_err("Failed to fetch the widgets state from SocialDB")?;
+                    })
+                    .wrap_err("Internal error: could not serialize SocialDB input args")?
+                    .into_bytes();
+
+                    calls.push(runtime.spawn(get_widgets(network_config.clone(), near_social_account_id.clone(), args)));
```

Comment:
Instead of spawning separately, I recommend using [`futures::stream::BufferUnordered`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffer_unordered). Here is a pseudocode:

```rust
let batch_size = 15;
let concurrency = 10;
let accounts = block_on(
    futures::stream::iter(local_widgets.keys().chunks(batch_size)).
        .flat_map(|batch_local_widget_names| fetch(...))
        .buffer_unordered(concurrency)
        .collect::<Result<HashMap<AccountId, SocialDbAccountMetadata>, _>>()
)?;
```

Check [this code](https://github.com/near/nearcore/blob/ab1c3b40277a095d367b42b18497a5b76bb82849/chain/rosetta-rpc/src/utils.rs#L364-L380) for inspiration.

Reply:
Yes, definitely a nicer approach. I'll incorporate it.